packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTable/TransactionsTableDepositRow.tsx

```diff
@@ -222,32 +234,65 @@ export function TransactionsTableDepositRow({
     [l1.network, tx.asset, tx.tokenAddress]
   )
 
+  const customAddressTxPadding = useMemo(
+    () => (isCustomAddressTx(tx) ? 'pb-10' : ''),
+    [tx]
+  )
+
+  if (!tx.sender || !address) {
+    return null
+  }
+
+  // both sent and received PENDING txs are stored together
+  // here we make sure we display a correct tx (sent or received)
+  if (isTransactionHistoryShowingSentTx) {
+    if (tx.sender.toLowerCase() !== address.toLowerCase()) {
+      return null
+    }
+  } else {
+    if (tx.sender.toLowerCase() === address.toLowerCase()) {
+      return null
+    }
+  }
+
   return (
     <tr
-      className={`text-sm text-dark ${
-        bgClassName || `bg-cyan even:bg-white`
-      } ${className}`}
+      className={twMerge(
+        'relative text-sm text-dark',
+        bgClassName || 'bg-cyan even:bg-white',
+        className
+      )}
       data-testid={`deposit-row-${tx.txId}`}
     >
-      <td className="w-1/5 py-3 pl-6 pr-3">
+      <td className={twMerge('w-1/5 py-3 pl-6 pr-3', customAddressTxPadding)}>
```

Comment:
Padding is added to each `td` (only for custom address) because the custom address label is position absolutely and needs some space within `tr`.